### PR TITLE
Add Sinopia indexer application and container

### PR DESCRIPTION
**Pull request by mjgiarlo**
connects to #46
connects to #47
connects to LD4P/sinopia_indexing_pipeline#1

Includes:

* Add a `sinopia_indexer` folder with a simple JavaScript application, ported over from `sinopia_client`, that:
  * Connects to stomp to listen for messages from Trellis
  * Parses (Trellis) URIs out of stomp messages
  * Hits Trellis API to get JSON-LD representation of a changed Trellis resource
  * Indexes the JSON in ElasticSearch
* Containerize indexer application (using dockerize to wait on dependencies)
* Add indexer service to docker-compose configuration
* Prefer `depends_on` to no-longer-recommended `links` declarations in `docker-compose` config
* Add sinopia_client's `package-lock.json`
* Corrects `license` field and adds `repository` field to both `package-lock.json` files to get rid of warnings on `npm install`
* Ignore `node_modules` wherever it appears in the codebase

Needs:

* Test coverage
* README documentation
* One or two refactorings of the application (to be more class-oriented and less procedural)
* Add es6 linter/beautifier?
* More sophisticated indexing—I suspect we shouldn't be indexing the entire JSON-LD document for a Trellis resource
* Possibly: a change in how the indexer interacts w/ Trellis. Current assumption is that the activity stream messages will contain a Trellis resource URI that the indexer can `GET` without knowledge of the `sinopia_server`'s internal workings

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -13,13 +13,20 @@ services:
     ports:
       - 8080:8080
       - 8081:8081
-    links:
-      - database
-      - broker
     depends_on:
```
Comment (jermnelson) [inline docker-compose.yml]: 👍 Just learning about `depends_on` in this commit

File: sinopia_indexer/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    stompClient.subscribe(queueName, (body, headers) => {
+      console.log(`received message with body: ${body}`)
+      uri = JSON.parse(body).object.id
+      console.log(`uri needs indexing: ${uri}`)
+      superagent.get(uri)
+        .accept('application/ld+json')
+        .then(res => {
+          json = res.body
+          console.log(`indexing trellis resource: ${json}`)
+          let indexResult = searchClient.index({
```
Comment (jermnelson) [inline sinopia_indexer/index.js]: If the indexResult is unsuccessful; would we send it to console.error or somewhere else? (Do you know if AWS fargate containers would have their logs available?)

File: sinopia_indexer/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "sinopia_indexer",
+  "version": "0.1.0",
+  "description": "Sinopia Indexer listens for STOMP messages, parses URIs out of messages, retrieves resources from an LDP server, and indexes them into ElasticSearch ",
+  "license": "Apache-2.0",
```
Comment (jermnelson) [inline sinopia_indexer/package.json]: You should add a `contributors` hash with your name!

File: sinopia_indexer/index.js (hunk 2 above)
Comment (mjgiarlo) [reply sinopia_indexer/index.js]: @jermnelson I think `console.error` is the simplest choice here. I suspect the errors would be automagically directed to CloudWatch Logs, but we ought to verify this. (It was at least true when writing golang code in Lambdas that output from the built-in `log` package went to CW Logs.) Error handling is another thing that needs to be added to this branch. I would do that in the course of writing up the test suite.

File: sinopia_indexer/package.json (hunk 3 above)
Comment (mjgiarlo) [reply sinopia_indexer/package.json]: Will do!

File: sinopia_indexer/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 Stanford University see Apache2.txt for license
+
+const elasticsearch = require('elasticsearch')
+const Stomp = require('stomp-client')
+const superagent = require('superagent')
+
+const searchHost = process.env.SEARCH_HOST || 'localhost'
+const brokerHost = process.env.BROKER_HOST || 'localhost'
```
Comment (jmartin-sul) [inline sinopia_indexer/index.js]: oh cool.  i recall us looking for a way to get at env vars when @jermnelson and i were pairing w/ sarav on the cognito work he did before he left.  can't recall that context well enough to remember whether we could use the same approach there, but prob a useful thing to keep in mind as a tool anyway.

File: sinopia_indexer/index.js (hunk 2 above)
Comment (jmartin-sul) [reply sinopia_indexer/index.js]: > I suspect the errors would be automagically directed to CloudWatch Logs, but we ought to verify this. (It was at least true when writing golang code in Lambdas that output from the built-in `log` package went to CW Logs.)

that would be cool if we got that for free.  i imagine that's an easier thing for amazon to wire up in a lambda, where they're more directly controlling the container that runs the code (which seems like it'd give tighter integration by default with the language and its packages).  but this is all new enough to me that i would be totally unsurprised if that sort of magic did exist.

Comment (mjgiarlo) [reply sinopia_indexer/index.js]: confirmed with @aaron-collier today that as long as the Fargate service/container is wired up to use AWS logging, STDOUT/STDERR from the container should flow to CW Logs.

Comment (jermnelson) [reply sinopia_indexer/index.js]: 👍

File: sinopia_indexer/index.js (hunk 4 above)
Comment (jermnelson) [reply sinopia_indexer/index.js]: Hi @jmartin-sul , the [.dotenv](https://www.npmjs.com/package/dotenv) npm module is the one you where thinking of that Sarav was using.

**Comment by jermnelson** [comment]
Hi @mjgiarlo , I like how you decomposed the script into separate javascript modules; very much the practice in React and Redux. I would suggest using [Jest](https://jestjs.io/) for your unit tests when you get to that point (if you haven't already).

File: sinopia_indexer/index.js (hunk 2 above)
Comment (jmartin-sul) [reply sinopia_indexer/index.js]: that's great!

**Pull request closed without merging**